src/java.base/share/classes/java/time/temporal/WeekFields.java

```diff
@@ -330,8 +330,10 @@ public static WeekFields of(DayOfWeek firstDayOfWeek, int minimalDaysInFirstWeek
         WeekFields rules = CACHE.get(key);
         if (rules == null) {
```

Comment:
Perhaps you don't even need this `if` block and the preceding `CACHE.get(key)` and maybe it can be replaced with:

```
return CACHE.computeIfAbsent(key, ignore -> new WeekFields(firstDayOfWeek, minimalDaysInFirstWeek));
```

Reply:
But it will generate garbage: non-static lambda.

Reply:
It already generates some garbage as it does string concatenation for the key. Here's an idea: declare a record class for the key, `record CacheKey(DayOfWeek firstDayOfWeek, int minimalDaysInFirstWeek)`. It will be more efficient than using strings for keys, and then you can have a static lambda.

Reply:
done